app/components/shared/join-banner.vue

```diff
@@ -18,7 +21,7 @@ defineSlots<{ default: string; title: string }>()
       variant="soft"
       color="warning"
       icon="ph:sparkle-duotone"
-      to="/login"
+      :to="`/login?redirect=${encodeURIComponent(currentPath)}`"
```

Comment:
I found this approach is buggy while maintaining existing query like if you in the `/reading-habits?date=05%2F24%2F2025` page. I encourage you to read the documentation, which we can use query object instead: https://ui.nuxt.com/components/link
![image](https://github.com/user-attachments/assets/0c52d2d4-c496-4ea4-a112-cd822f14b78c)